### DISCUSSION TITLE
Any examples of how to read a full KML file?

### DISCUSSION THREAD

**Discussion by lorenh** [NetTopologySuite/NetTopologySuite#679] [Q&A] [answered]
Given any KML file, can I use NetTopologySuite to read it? Are there any examples I can look at?  If NTS doesn't handle it robustly, are there other libraries to do so?

Thanks!

**Comment by FObermaier** [comment, marked as answer]
NTS' KMLReader __only__ parses the geometry part and enclosed attributes. The usage is demonstrated in [unit tests](https://github.com/NetTopologySuite/NetTopologySuite/blob/develop/test/NetTopologySuite.Tests.NUnit/IO/KML/KMLReaderTest.cs).

Have you tried [SharpKml.Core](https://www.nuget.org/packages/SharpKml.Core)
